File: Projeto_CodeDeveloper_24/Models/Receitas.cs
using System.Text.Json.Serialization;

namespace Projeto_CodeDeveloper_24.Models
{
    public class Receitas
    {
        public int Id { get; set; }
        public string? Titulo { get; set; }
        public int? Duracao { get; set; }

        public string? Dificuldade { get; set; }
        public string? Descricao { get; set; }
        

        // public List<Ingredientes>? Ingredientes { get; } = [];  //FK
        public List<ReceitaIngredientes>? ReceitaIngredientes { get; set; } = [];

        public int CategoriasId { get; set; }

        [JsonIgnore] public Categorias? Categorias { get; set; }
    }
}

File: src/app/models/receitas.ts
import { Categorias } from './categorias';
import { ReceitaIngredientes } from './receitaIngredientes';

export interface Receitas {
  id: number | null;
  titulo: string;
  duracao: number;
  dificuldade: string;
  categoria: Categorias;
  descricao: string;
  receitaIngredientes: ReceitaIngredientes[];
}
